ada_feeding/ada_feeding/behaviors/moveit2/moveit2_plan.py

```diff
@@ -0,0 +1,641 @@
+    Or SUCCESS + None if the goal constraints are all satisfied.
+
+    Or FAILURE + None if no trajectory can be found.
+    """
+
+    # pylint: disable=arguments-differ
+    # We *intentionally* violate Liskov Substitution Princple
+    # in that blackboard config (inputs + outputs) are not
+    # meant to be called in a generic setting.
+
+    # pylint: disable=too-many-arguments
+    # These are effectively config definitions
+    # They require a lot of arguments.
+
+    def blackboard_inputs(
+        self,
+        goal_constraints: Union[
+            BlackboardKey, List[Tuple[MoveIt2ConstraintType, Dict[str, Any]]]
+        ],
+        path_constraints: Optional[
+            Union[BlackboardKey, List[Tuple[MoveIt2ConstraintType, Dict[str, Any]]]]
+        ] = None,
+        ignore_violated_path_constraints: Union[BlackboardKey, bool] = False,
+        pipeline_id: Union[BlackboardKey, str] = "ompl",
+        planner_id: Union[BlackboardKey, str] = "RRTstarkConfigDefault",
+        allowed_planning_time: Union[BlackboardKey, float] = 0.5,
+        max_velocity_scale: Union[BlackboardKey, float] = 0.1,
+        max_acceleration_scale: Union[BlackboardKey, float] = 0.1,
+        cartesian: Union[BlackboardKey, bool] = False,
+        cartesian_max_step: Union[BlackboardKey, float] = 0.0025,
+        cartesian_jump_threshold: Union[BlackboardKey, float] = 0.0,
+        cartesian_fraction_threshold: Union[BlackboardKey, float] = 0.0,
+        start_joint_state: Union[
+            BlackboardKey, Optional[Union[JointState, List[float]]]
+        ] = None,
+        debug_trajectory_viz: Union[BlackboardKey, bool] = False,
+    ) -> None:
+        """
+        Blackboard Inputs
+
+        Parameters
+        ----------
+        goal_constraints: list of constraints, each one is a pair mapping the type to the kwargs
+                        passed to the corresponding pymoveit2 function. See moveit2.py:set_X_goal()
+        path_constraints: list of constraints, each one is a pair mapping the type to the kwargs
+                        passed to the corresponding pymoveit2 function.
+                        See moveit2.py:set_path_X_constraint()
+        ignore_violated_path_constraints: if True, if a path constraint starts violated,
+                                          it's okay to plan without it.
+        pipeline_id: MoveIt2 Pipeline, usually "ompl"
+        planner_id: MoveIt2 planner to use
+        allowed_planning_time: Anytime planner timeout, passed to MoveIt2 Directly.
+                                NOTE: Not guaranteed, use a Timeout decorator for that.
+        max_velocity_scale: [0,1] determines max joint of the trajectory
+        max_acceleration_scale: [0,1] determines max joint acceleration of the trajectory
+        cartesian: True to use MoveIt2's cartesian planner, uses the below parameters
+        cartesian_max_step: % of the geodesic to step for each IK of the cartesian planner
```

Comment:
Yeah, looks like meters, but since it is only a single value, for rotations, steps there are 2deg/1cm.

https://github.com/ros-planning/moveit2/blob/498a9f3d659cbcf0db03cef584d4ca16468e4dab/moveit_core/robot_state/include/moveit/robot_state/cartesian_interpolator.h#L79